Dockerfile.soroban-rpc

```diff
@@ -0,0 +1,21 @@
+FROM golang:1.19 AS builder
+
+ARG REF
+WORKDIR /go/src/github.com/stellar/soroban-tools
+RUN git clone https://github.com/stellar/soroban-tools /go/src/github.com/stellar/soroban-tools
+RUN git fetch origin ${REF}
+RUN git checkout ${REF}
+RUN git config --global --add safe.directory "/go/src/github.com/stellar/soroban-tools"
+
+ENV CARGO_HOME=/rust/.cargo
+ENV RUSTUP_HOME=/rust/.rust
+ENV PATH="/usr/local/go/bin:$CARGO_HOME/bin:${PATH}"
+ENV DEBIAN_FRONTEND=noninteractive
+RUN apt-get update
+RUN apt-get install -y build-essential
+RUN apt-get clean
+
+RUN curl https://sh.rustup.rs -sSf | sh -s -- -y --default-toolchain stable
+
+RUN make build-soroban-rpc
+RUN mv soroban-rpc /bin/soroban-rpc
```

Comment:
The goal is to keep the dockerfiles in the respective repos, so ideally we'd update the dockerfile in the soroban-rpc repo so that it can successfully build the entire product instead of duplocstikng build process here. 

Is there a reason we need the file here?

(Note that we do have a horizon file here, but that's only because we couldn't update the horizon file for a past release. Well switch back to the horizon repos file in a future version of quickstart.)

Reply:
yes, I saw the Horizon dockerfile, and seemed like a pattern to follow, @tsachiherman mentioned same - https://github.com/stellar/quickstart/pull/415#issuecomment-1409548729

so, looks like I'll fix this from the soroban-tools side instead, closing this pr then.

Reply:
did the equivalent change on the soroban-tools dockerfile - https://github.com/stellar/soroban-tools/pull/371